File: DSA/VaniAgrawal_590015846/Day_29_Question_2.py
# Extract and check the value of a specific bit position using efficient bitwise masking operations.

n = int(input("Enter the number (n): "))
i = int(input("Enter the bit position to check (i): "))

if (n & (1 << i)) != 0:
    print(1)
else:
    print(0)

File: DSA/Ayush_Mishra_590014844/Day_29_Question_2.c
#include <stdio.h>

int main() {
    int n, i;
    printf("Enter the number (n): ");
    scanf("%d", &n);
    printf("Enter the bit position (i): ");
    scanf("%d", &i);
    int bit = (n >> i) & 1;
    printf("The %d-th bit is: %d\n", i, bit);
    return 0;
}
